### PR TITLE
update NativeInterop and WebView2 package versions

### LINKED ISSUE #4495: [Bug] some net6.0-windows dependencies are built with optimizations disabled
### Library version used

4.58.1

### .NET version

.net6.0

### Scenario

Other - please specify

### Is this a new or an existing app?

The app is in production, and I have upgraded to a new version of MSAL

### Issue description and reproduction steps

I updated `Microsoft.Data.SqlClient` from `5.1.2` to `5.1.4` and it updated its dependency on `Microsoft.Identity.Client` from `4.47.2` to `4.56.0`, which added additional dependencies for `net6.0-windows` (`Microsoft.Identity.Client.NativeInterop` and `Microsoft.Web.WebView2`).

Both of these have `DisableOptimizations` set in their `DebuggableAttribute`.

This also occurs on the versions used by `Microsoft.Identity.Client` version `4.58.1` (current latest version)

Is this intentional?

I have a check in my CI pipeline to stop debug assemblies from being used. I can exclude these if it's intentional, but just wanted to check if it was intentional or accidental.

### Relevant code snippets

_No response_

### Expected behavior

_No response_

### Identity provider

Microsoft Entra ID (Work and School accounts and Personal Microsoft accounts)

### Regression

_No response_

### Solution and workarounds

_No response_

### PR DESCRIPTION
**Pull request by SimonCropp**
* Microsoft.Identity.Client.NativeInterop 0.14.1
 * Microsoft.Web.WebView2 to 1.0.2210.55

### CODE REVIEW

**Comment by SimonCropp** [comment]
hmm seems to only fix the debuggable attributes in Microsoft.Web.WebView2. 

Microsoft.Identity.Client.NativeInterop  is still
```
[assembly: Debuggable(
DebuggableAttribute.DebuggingModes.Default |
DebuggableAttribute.DebuggingModes.DisableOptimizations |
DebuggableAttribute.DebuggingModes.IgnoreSymbolStoreSequencePoints |
DebuggableAttribute.DebuggingModes.EnableEditAndContinue)]

```

File: Directory.Packages.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
     <!-- Version of the Microsoft.Identity.Client.NativeInterop package. -->
-    <MSALRuntimeNativeInteropVersion>0.13.14</MSALRuntimeNativeInteropVersion>
+    <MSALRuntimeNativeInteropVersion>0.14.1</MSALRuntimeNativeInteropVersion>
```
Comment (bgavrilMS) [inline Directory.Packages.props:5]: @ashok672 - is this ok?

CC @gladjohn

**Review by bgavrilMS — APPROVED** [review]

File: Directory.Packages.props (hunk 1 above)
Comment (SimonCropp) [reply Directory.Packages.props:5]: also does anyone what repo Microsoft.Identity.Client.NativeInterop comes from?

**Review by pmaytak — APPROVED** [review]

File: Directory.Packages.props (hunk 1 above)
Comment (bgavrilMS) [reply Directory.Packages.props:5]: It's a non-public repo. What changes would we have to make there?

Comment (SimonCropp) [reply Directory.Packages.props:5]: according to this https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4495 it has DisableOptimizations set in their DebuggableAttribute.

Comment (gladjohn) [reply Directory.Packages.props:5]: I would let @ashok672 confirm this

**Comment by bgavrilMS** [comment]
> hmm seems to only fix the debuggable attributes in Microsoft.Web.WebView2.
> 
> Microsoft.Identity.Client.NativeInterop is still
> 
> ```
> [assembly: Debuggable(
> DebuggableAttribute.DebuggingModes.Default |
> DebuggableAttribute.DebuggingModes.DisableOptimizations |
> DebuggableAttribute.DebuggingModes.IgnoreSymbolStoreSequencePoints |
> DebuggableAttribute.DebuggingModes.EnableEditAndContinue)]
> ```

I didn't find any reference to this attribute in the interop project. However, the docs state that: 

`This enumeration is primarily used by language developers. It is generally not used in application development. Development environments use [DebuggableAttribute.DebuggingModes](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.debuggableattribute.debuggingmodes?view=net-8.0) based on compiler parameters such as /debug and /optimize.`

So I guess that when we build the interop (it's a non-standard build), we don't add the correct flag.

**Comment by SimonCropp** [comment]
@ashok672 bump

File: Directory.Packages.props (hunk 1 above)
Comment (SimonCropp) [reply Directory.Packages.props:5]: @ashok672 bump

**Comment by bgavrilMS** [comment]
Should be ok to take this, PCA folks will be updating the package soon anyway.

**Pull request closed without merging**